src/GraphQL/Resolver/TranslationListing.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\DataHubBundle\GraphQL\Resolver;
+
+use GraphQL\Type\Definition\ResolveInfo;
+use Pimcore\Bundle\DataHubBundle\GraphQL\ElementDescriptor;
+use Pimcore\Bundle\DataHubBundle\GraphQL\Service;
+use Pimcore\Bundle\DataHubBundle\GraphQL\Traits\ServiceTrait;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class TranslationListing
+{
+    use ServiceTrait;
+
+    protected EventDispatcherInterface $eventDispatcher;
+
+    public function __construct(Service $graphQlService, EventDispatcherInterface $eventDispatcher)
+    {
+        $this->setGraphQLService($graphQlService);
+
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    public function resolveEdge(mixed $value = null, array $args = [], array $context = [], ResolveInfo $resolveInfo = null): mixed
+    {
+        $translation = $value['node'];
+        $data = new ElementDescriptor();
+
+        $fieldHelper = $this->getGraphQlService()->getObjectFieldHelper();
+
+        return $fieldHelper->extractData($data, $translation, $args, $context, $resolveInfo);
+    }
+
+    public function resolveEdges(mixed $value = null, array $args = [], array $context = [], ResolveInfo $resolveInfo = null): mixed
+    {
+        return $value['edges'];
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function resolveListing(mixed $value = null, array $args = [], array $context = [], ResolveInfo $resolveInfo = null): array
+    {
+        // get list of types
+        $list = new \Pimcore\Model\Translation\Listing();
+
+        if (!empty($args['keys'])) {
+            $keysArray = explode(',', $args['keys']);
+            $keysString = "'" . implode("','", $keysArray) . "'" ;
+            $list->setCondition('translations_messages.key IN (' . $keysString . ')');
+        }
+
+        if (!empty($args['domain'])) {
+            $list->setDomain($args['domain']);
+        }
+
+        // sorting
+        if (!empty($args['sortBy'])) {
+            $list->setOrderKey($args['sortBy']);
+            if (!empty($args['sortOrder'])) {
+                $list->setOrder($args['sortOrder']);
+            }
+        }
+
+        // paging
+        if (isset($args['first'])) {
+            $list->setLimit($args['first']);
+        }
+
+        if (isset($args['after'])) {
+            $list->setOffset($args['after']);
+        }
+
+        $totalCount = $list->getTotalCount();
```

Comment:
```suggestion

 $event = new ListingEvent(
            $list,
            $args,
            $context,
            $resolveInfo
        );
        $this->eventDispatcher->dispatch($event, ListingEvents::PRE_LOAD);
        $list = $event->getListing();

        $totalCount = $list->getTotalCount();
```
Since we have added the eventDispatcher already, we could use it hear to trigger a listing event